### PR TITLE
issue #1242 fix: non existent file directives

### PR DESCRIPTION
**Pull request by pbhatnagar3**
#### Manual Testing

For testing, the following worksheet was used:

```
***display table directive***

% schema s1
% add uuid
% add name
% add output /output
% display table s1
[dataset graph-1.txt]{0x440cfb819ef5412d90bc6e482eef79f1}
[dataset graph-2.txt]{0x86e03564a45d40d6a802482f87e47256}
[run topological-sort-run-graph-1 -- :topological-sort.py,input:graph-1.txt : python topological-sort.py < input > output]{0xa3817866bf2c4aeeb5f516546270b105}

% schema s2
% add uuid
% add output-1 /output-1
% display table s2
[run topological-sort-run-graph-1 -- :topological-sort.py,input:graph-1.txt : python topological-sort.py < input > output]{0xa3817866bf2c4aeeb5f516546270b105}


***display content directive***

% display contents /output
[dataset graph-2.txt]{0x86e03564a45d40d6a802482f87e47256}

% display contents /output
[run topological-sort-run-graph-1 -- :topological-sort.py,input:graph-1.txt : python topological-sort.py < input > output]{0xa3817866bf2c4aeeb5f516546270b105}

```

Before, when missing files were displayed using the `display contents /file` directive, an internal 500 error was thrown. 
image

Also, when a column, associated to a file that was not present, was added to table, '' (empty string corresponding to None) was shown in that column. 

image

NOTE: For the above screenshot, the following lines were removed from the worksheet to avoid getting the 500 error.
```
% display contents /output
[dataset graph-2.txt]{0x86e03564a45d40d6a802482f87e47256}
```
Now, when `display content /file` is used to display a non-existent file, 'MISSING' is shown. The same applies for table columns which are associated to files that do not yet exist.

Here is a screenshot showing the changes:
image

### CODE REVIEW

File: codalab/apps/web/bundles.py [hunk 1]
```diff
@@ -121,7 +121,16 @@ def worksheet(self, uuid, interpreted=False):
                 for item in worksheet_info['items']:
                     if item['mode'] in ['html', 'contents']:
                         # item['name'] in ['stdout', 'stderr']
-                        item['interpreted'] = map(base64.b64decode, item['interpreted'])
+                        try:
+                            item['interpreted'] = map(base64.b64decode, item['interpreted'])
+                        except Exception, e:
```
Comment (percyliang) [inline codalab/apps/web/bundles.py]: Catching an exception is a bit blunt - can you just test if things are None?

File: codalab/apps/web/bundles.py [hunk 2]
```diff
@@ -121,7 +121,16 @@ def worksheet(self, uuid, interpreted=False):
                 for item in worksheet_info['items']:
                     if item['mode'] in ['html', 'contents']:
                         # item['name'] in ['stdout', 'stderr']
-                        item['interpreted'] = map(base64.b64decode, item['interpreted'])
+                        if item['interpreted'] is None:
+                            item['interpreted'] = ['MISSING']
+                        else:
+                            item['interpreted'] = map(base64.b64decode, item['interpreted'])
+                    elif item['mode'] == 'table':
```
Comment (percyliang) [inline codalab/apps/web/bundles.py]: Remove this section, because resolve_interpreted_items already returns 'MISSING', not None for tables.

**Comment by pbhatnagar3** [comment]
In the case of 
```
% display contents /output
[dataset graph-1.txt]{0x7ed7e8ba6b7b428aa4507dbc27d749b7}
```
 when file is not present, local_bundle_client.resolve_iterpreted_items returns None that crashes codalab code in ``codalab/apps/web/bundles.py``. This change handles that case.

**Comment by pbhatnagar3** [comment]
Have put out another pull request regarding the same. Closing this one.

**Pull request closed without merging**